Document client-side to server-side apply upgrade (#22547)
kubernetes/website: content/en/docs/reference/using-api/api-concepts.md

using server-side apply with the following flag.

```
kubectl apply --server-side [--dry-run=server]
```

By default, field management of the object will transfer from client-side apply
to server-side apply without encountering conflicts.

Note that we infer client-side apply's managed fields via the
`last-applied-configuration` annotation. It should be kept up-to-date. Any
fields not managed by client-side apply will raise conflicts. For example, if
`kubectl scale` was used to update the replicas field after client-side apply,
then this field will not be owned by client-side apply and will conflict on
`kubectl apply --server-side`.

As an exception, users can opt-out of this behavior by specifying a different,
non-default field manager. The default field manager for kubectl server-side
apply is `kubectl`.

```
kubectl apply --server-side --field-manager=my-manager [--dry-run=server]
```

### Downgrading from Server-Side Apply to Client-Side Apply

Server-side apply users who manage a resource with `kubectl apply --server-side`
may downgrade back to client-side apply directly with `kubectl apply`.

Downgrading works because server-side apply will keep the

> `last-applied-configuration` annotation up-to-date for users who may use

Reviewer: Only for `kubectl` managers?

Author: Oh yes only for manager == `kubectl`

Author: Updated

change the API version they are using, but Server Side Apply supports this use
case.

> ### Upgrading from Client-Side Apply to Server-Side Apply

Reviewer 2: ```suggestion
### Upgrading from client-side apply to server-side apply
```

> Client-side apply users who manage a resource with `kubectl apply` may start

Reviewer 2: ```suggestion
Client-side apply users who manage a resource with `kubectl apply` can start
```

using server-side apply with the following flag.

```
kubectl apply --server-side [--dry-run=server]
```

> By default, field management of the object will transfer from client-side apply

Reviewer 2: Avoid future tense.

```suggestion
By default, field management of the object transfers from client-side apply
```

to kubectl server-side apply without encountering conflicts.

> Note that we infer client-side apply's managed fields via the
> `last-applied-configuration` annotation. It should be kept up-to-date. Any
> fields not managed by client-side apply will raise conflicts. For example, if
> `kubectl scale` was used to update the replicas field after client-side apply,
> then this field will not be owned by client-side apply and will conflict on
> `kubectl apply --server-side`.

Reviewer 2: Avoid "we" and Latin abbreviations ("via"). 

Add a caution callout.

```suggestion
{{< caution >}}
Keep the `last-applied-configuration` annotation up to date. The annotation infers client-side apply's managed fields. Any fields not managed by client-side apply raise conflicts.

For example, if you used `kubectl scale` to update the replicas field after client-side apply,
then this field is not owned by client-side apply and creates conflicts on `kubectl apply --server-side`.
{{< /caution >}}
```

This behavior applies to server-side apply with the `kubectl` field manager. As

> an exception, users can opt-out of this behavior by specifying a different,

Reviewer 2: Use "you".

```suggestion
an exception, you can opt-out of this behavior by specifying a different,
```

> non-default field manager, as seen below. The default field manager for kubectl

Reviewer 2: ```suggestion
non-default field manager, as seen in the following example. The default field manager for kubectl
```

server-side apply is `kubectl`.

```
kubectl apply --server-side --field-manager=my-manager [--dry-run=server]
```

> ### Downgrading from Server-Side Apply to Client-Side Apply

Reviewer 2: ```suggestion
### Downgrading from server-side apply to client-side apply
```

> Server-side apply users who manage a resource with `kubectl apply --server-side`

Reviewer 2: ```suggestion
If you manage a resource with `kubectl apply --server-side`,
```

> may downgrade back to client-side apply directly with `kubectl apply`.

Reviewer 2: ```suggestion
you can downgrade to client-side apply directly with `kubectl apply`.
```

> Downgrading works because kubectl server-side apply will keep the

Reviewer 2: ```suggestion
Downgrading works because kubectl server-side apply keeps the
```

> `last-applied-configuration` annotation up-to-date for users who may use

Reviewer 2: ```suggestion
`last-applied-configuration` annotation up-to-date if you use
```

`kubectl apply`.

This behavior applies to server-side apply with the `kubectl` field manager. As

> an exception, users can opt-out of this behavior by specifying a different,

Reviewer 2: ```suggestion
an exception, you can opt-out of this behavior by specifying a different,
```

> non-default field manager, as seen below. The default field manager for kubectl

Reviewer 2: ```suggestion
non-default field manager, as seen in the following example. The default field manager for kubectl
```
